src/components/shared/Button/Button.tsx

```diff
@@ -0,0 +1,58 @@
+import clsx from "clsx";
+import Link from "next/link";
+import ButtonIcon from "./ButtonIcon";
+import { IconName } from "@/types";
+
+export interface ButtonProps {
```

Comment:
To make the component even more reusable we can accept all `<button>` properties:
```
interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> { ... }
```